runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ConfigBuilder.java

```diff
@@ -111,7 +111,7 @@ public Config build() {
 
       validateConfigs(options, config);
 
-      return new MapConfig(config);
+      return new HashMap<>(config);
```

Comment:
The `config` is already a HashMap, we don't need `new HashMap<>` again.
```
  private final Map<String, String> config = new HashMap<>();
```

Reply:
Some part of the code might return a ImmutableMap, so this is to prevent the output being non-deterministic. The listener will always have a mutable config map to update.